shared/src/state-migrations/42-replace-clientroles.ts

```diff
@@ -0,0 +1,43 @@
+import type { UUID } from '../utils/index.js';
+import type { Migration } from './migration-functions.js';
+
+type OldRole = 'participant' | 'trainer';
+
+interface ClientRole {
+    type: 'clientrole';
+    mainRole: 'participant' | 'trainer';
+    specificRole: 'map-operator' | 'trainer';
+}
```

Comment:
Keep in mind to update those values to camelCasing, after changing the main type definitions:

```suggestion
    type: 'clientRole';
    mainRole: 'participant' | 'trainer';
    specificRole: 'mapOperator' | 'trainer';
}
```